File: Easy/Power of Thor/Python 3/Power of Thor.py
import sys

light_x, light_y, initial_tx, initial_ty = [int(i) for i in input().split()]

while 1:

    if light_x > initial_tx:
        initial_tx += 1
        directionX = "E"
    elif light_x < initial_tx:
        initial_tx -= 1
        directionX = "W"
    else:
        directionX = ""
    
    if light_y > initial_ty:
        initial_ty += 1
        directionY = "S"
    elif light_y < initial_ty:
        initial_ty -= 1
        directionY = "N"
    else:
       directionY = ""
        
    print(directionY + directionX)

File: Easy/Power of Thor/Ruby/PowerOfThor.rb
STDOUT.sync = true

@light_x, @light_y, @initial_tx, @initial_ty = gets.split(" ").collect {|x| x.to_i}

loop do
  if @light_x > @initial_tx
      @initial_tx += 1
      @direction_X = "E"
  elsif @light_x < @initial_tx
      @initial_tx -= 1
      @direction_X = "W"
  else
      @direction_X = ""
  end

  if @light_y > @initial_ty
      @initial_ty += 1
      @direction_Y = "S"
  elsif @light_y < @initial_ty
      @initial_ty -= 1
      @direction_Y = "N"
  else
      @direction_Y = ""
  end

  puts (@direction_Y + @direction_X)
end